src/components/Segmented/Segmented.test.tsx

```diff
@@ -0,0 +1,199 @@
+
+    const props = {
+      defaultValue: selectedOption,
+      options: stringOptions,
+      onChange,
+    }
+
+    test('renders options correctly', () => {
+      const { asFragment } = render(<Segmented {...props} />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+
+      stringOptions.forEach(name => {
+        const option = screen.getByRole('listitem', { name })
+
+        expect(option).toBeVisible()
+        expect(option).toHaveAttribute('aria-disabled', 'false')
+      })
+
+      expect(screen.getByRole('listitem', { name: selectedOption })).toHaveAttribute('aria-checked', 'true')
+
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    test('renders vertical options correctly', () => {
+      const { asFragment } = render(<Segmented {...props} optionsAlignment={Vertical} />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+
+      stringOptions.forEach(name => {
+        const option = screen.getByRole('listitem', { name })
+
+        expect(option).toBeVisible()
+        expect(option).toHaveAttribute('aria-disabled', 'false')
+      })
+
+      expect(screen.getByRole('listitem', { name: selectedOption })).toHaveAttribute('aria-checked', 'true')
+
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    test('renders disabled segmented correctly', () => {
+      const { asFragment } = render(<Segmented {...props} isDisabled />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+
+      stringOptions.forEach(name => {
+        const option = screen.getByRole('listitem', { name })
+
+        expect(option).toBeVisible()
+        expect(option).toHaveAttribute('aria-disabled', 'true')
+      })
+
+      expect(screen.getByRole('listitem', { name: selectedOption })).toHaveAttribute('aria-checked', 'true')
+
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    test('select numeric default value correctly', () => {
+      render(<Segmented {...props} defaultValue={2} />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+      expect(screen.getByRole('listitem', { name: selectedOption })).toHaveAttribute('aria-checked', 'true')
+    })
+
+    test('select value correctly', () => {
+      render(<Segmented {...props} value={controlledOption} />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+      expect(screen.getByRole('listitem', { name: controlledOption })).toHaveAttribute('aria-checked', 'true')
+    })
+
+    test('select numeric value correctly', () => {
+      render(<Segmented {...props} value={4} />)
+
+      expect(screen.getByRole('list')).toBeVisible()
+      expect(screen.getByRole('listitem', { name: controlledOption })).toHaveAttribute('aria-checked', 'true')
+    })
+
+    test('calls onChange correctly', () => {
+      render(<Segmented {...props} />)
+
+      fireEvent.click(screen.getByRole('listitem', { name: clickedOption }))
+
+      expect(onChange).toBeCalledTimes(1)
+      expect(onChange).toBeCalledWith(clickedOption)
+    })
+  })
+})
```

Comment:
```suggestion
})

```

Reply:
Can we set up an ESLint rule to automatically place one line at the end of the file? 🤔

Reply:
I think it's a common best practice to have one and only empty line at the end of the file (GitLab itself advises you when it's missing, saying `\ No newline at end of file`)

Reply:
Maybe I misunderstood the first time, it should be already the way you are saying. I thought you meant an empty line and a new line

Reply:
You are right, if I try to delete the last line I get the error `Newline required at end of file but not found.` and saving it is automatically added. Can it be GitHub that doesn't let me see it then? 🤔

Reply:
Yes, I think Github is purposely removing the last newline of each file in its UI.